tests/test_trame.py

```diff
@@ -0,0 +1,660 @@
+# ------------------------------------------------------------------
+# Copyright (c) 2020 PyInstaller Development Team.
+#
+# This file is distributed under the terms of the GNU General Public
+# License (version 2.0 or later).
+#
+# The full license is available in LICENSE, distributed with
+# this software.
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+# ------------------------------------------------------------------
+
+from PyInstaller.utils.tests import importorskip
+
+
+@importorskip("trame")
+def test_trame(pyi_builder):
+    pyi_builder.test_source("""
+        import trame
+    """)
+
+
+@importorskip('trame_client')
+def test_trame_client(pyi_builder):
+    pyi_builder.test_source("""
+        import asyncio
+
+        from trame.app import get_server
+
+
+        async def stop(*args, **kwargs):
+            await server.stop()
+
+
+        server = get_server()
+        server.controller.on_server_ready.add(
+            lambda *args, **kwargs: asyncio.ensure_future(stop(*args, **kwargs))
+        )
+        server.start(port=0, open_browser=False)
+    """)
+
+
+@importorskip('trame_vuetify')
+def test_trame_vuetify(pyi_builder):
+    pyi_builder.test_source("""
+        import asyncio
+
+        from trame.app import get_server
+        from trame.ui.vuetify3 import VAppLayout
+
+
+        async def stop(*args, **kwargs):
+            await server.stop()
+
+
+        server = get_server()
+        VAppLayout(server)
+        server.controller.on_server_ready.add(
+            lambda *args, **kwargs: asyncio.ensure_future(stop(*args, **kwargs))
+        )
+        server.start(port=0, open_browser=False)
+    """)
+
+
+@importorskip("vtkmodules")
+@importorskip("trame_vtk")
+def test_trame_vtk(pyi_builder):
+    pyi_builder.test_source("""
+        import asyncio
+
+        from trame.app import get_server
+        from trame.ui.html import DivLayout
+        from trame_vtk.widgets import vtk
+
+
+        async def stop(*args, **kwargs):
+            await server.stop()
+
+
+        server = get_server()
+        with DivLayout(server):
+            vtk.VtkMesh("test")
+        server.controller.on_server_ready.add(
+            lambda *args, **kwargs: asyncio.ensure_future(stop(*args, **kwargs))
+        )
+        server.start(port=0, open_browser=False)
+    """)
+
+
+@importorskip("pyvista")
+@importorskip("vtkmodules")
+@importorskip("nest_asyncio")
+@importorskip("trame_vtk")
+def test_trame_vtk_tools(pyi_builder):
+    pyi_builder.test_source("""
+        import os
+
+        import pyvista as pv
+        import trame_vtk
+
+        plotter = pv.Plotter()
+        plotter.export_html("test.html")  # Uses trame_vtk
+        os.remove("test.html")
```

Comment:
Can we use pytest's `tmp_path` fixture here and put the exported file under there?